services/core-api/app/api/projects/project_summary/models/project_summary.py

```diff
@@ -282,54 +342,33 @@ def update(self,
-
-            # if updates are made to the existing party:
-            if self.agent_party_guid is not None and str(self.agent_party_guid) == agent_party_guid:
-                Party.validate_phone_no(agent.get('phone_no'), address.get('address_type_code'))
-                self.agent.deep_update_from_dict(agent)
-                agent_address = Address(
-                    suite_no=address.get('suite_no'),
-                    address_line_1=address.get('address_line_1'),
-                    city=address.get('city'),
-                    sub_division_code=address.get('sub_division_code'),
-                    post_code=address.get('post_code'),
-                    address_type_code=address.get('address_type_code'),
-                )
-
-                self.agent.address.append(agent_address)
-                self.agent.save()
-            else:
-                agent_party = Party.create(
-                    party_name=agent.get('party_name'),
-                    first_name=agent.get('first_name'),
-                    phone_no=agent.get('phone_no'),
-                    phone_ext=agent.get('phone_ext'),
-                    email=agent.get('email'),
-                    party_type_code=agent.get('party_type_code'),
-                    job_title=agent.get('job_title'),
-                    job_title_code='AGT',
-                    address_type_code=address.get('address_type_code')
-                )                
-                agent_address = Address.create(
-                    suite_no=address.get('suite_no'),
-                    address_line_1=address.get('address_line_1'),
-                    city=address.get('city'),
-                    sub_division_code=address.get('sub_division_code'),
-                    post_code=address.get('post_code'),
-                    address_type_code=address.get('address_type_code'),
-                )
-                agent_party.address.append(agent_address)
-                agent_party.save()
-                self.agent_party_guid = agent_party.party_guid
+            agent_party = self.create_or_update_party(agent, 'AGT', self.agent)
+            agent_party.save()
+            self.agent_party_guid = agent_party.party_guid
+
+        self.facility_type = facility_type
+        self.facility_desc = facility_desc
+        self.facility_latitude = facility_latitude
+        self.facility_longitude = facility_longitude
+        self.facility_coords_source = facility_coords_source
+        self.facility_coords_source_desc = facility_coords_source_desc
+        self.facility_pid_pin_crown_file_no = facility_pid_pin_crown_file_no
+        self.legal_land_desc = legal_land_desc
+        self.facility_lease_no = facility_lease_no
+        self.zoning = zoning
+        self.zoning_reason = zoning_reason
+
+        if facility_operator:
+            if not facility_operator['party_type_code']:
+                facility_operator["party_type_code"] = "PER"
```

Comment:
Do we need an enum for party_type_code?

Reply:
This one only has two possible values, PER and ORG, so I dunno, enum seems overkill to me. Codebase is indeed littered with literals for this field.